src/document-converter.ts

```diff
@@ -86,6 +86,19 @@ const legacyJavascriptTypes: ReadonlySet<string|null> = new Set([
   'text/x-javascript',
 ]);
 
+/**
+ * This is a set of JavaScript files that we know to be converted as modules.
+ * This is neccessary because we don't definitively know the type of an external
+ * JavaScript file loaded in to a top-level HTML file.
```

Comment:
We know the type on in the input - that's completely determined by the `module` attribute. I think you're saying we don't know they type of the output, yeah?

Reply:
good point, I'll clarify